metricflow/model/data_warehouse_model_validator.py

```diff
@@ -130,46 +161,60 @@ def gen_dimension_tasks(cls, model: UserConfiguredModel) -> List[DataWarehouseVa
         on the data source to identify which have issues.
         """
 
+        render_tools = QueryRenderingTools(model=model, system_schema=system_schema)
+
         tasks: List[DataWarehouseValidationTask] = []
-        for index, data_source in enumerate(model.data_sources):
-            # generate the subtasks
-            data_source_tasks: List[DataWarehouseValidationTask] = []
-            data_source_columns: List[str] = []
-            for dimension in data_source.dimensions:
-                dim_to_query = dimension.expr if dimension.expr is not None else dimension.name
-                data_source_columns.append(dim_to_query)
-
-                data_source_tasks.append(
+        for data_source in model.data_sources:
+            if len(data_source.dimensions) > 0:
+                source_node = cls._data_source_node(render_tools=render_tools, data_source=data_source)
```

Comment:
Never mind, I see what I missed. You might want this to unnest the below, but it's optional:
```suggestion
            if not data_source.dimensions:
                continue

            source_node = cls._data_source_node(render_tools=render_tools, data_source=data_source)
```